lib/twingly/url.rb

```diff
@@ -103,11 +108,27 @@ def try_addressable_normalize(addressable_uri)
         raise
       end
 
+      def valid_hostname?(hostname)
+        # No need to check the TLD, the public suffix list does that
+        labels = hostname.split(DOT)[0...-1].map(&:to_s)
```

Comment:
Need the `0` for JRuby 9.3: https://github.com/twingly/twingly-url/actions/runs/3192326669/jobs/5209668938#step:4:30